fd-beta-version-backend/src/v1/model/dbrestaurantmenu.ts

```diff
@@ -0,0 +1,39 @@
+import { appdb } from "./appdb";
+
+export class dbRestaurantMenu extends appdb {
+  constructor() {
+    super();
+    this.table = "restaurantmenu";
+    this.uniqueField = "id";
+  }
+
+  async getMenuWithFoodItems(restaurantId: number) {
+    const fields = `
+      fi.id AS foodItemId,
+      rm.menuname,
+      fi.item_name,
+      fi.price,
+      fi.description,
+      fi.isavailable
+    `;
+
+    this.where = `WHERE rm.restaurant_id =  ${restaurantId}`;
+    this.orderby = "";
+
+    const table = `
+      restaurantMenu rm
+      JOIN foodItems fi ON rm.id = fi.menu_id
+    `;
+
+    try {
+     
+      this.table = table; 
+      const result = await this.allRecords(fields);
+
+      return result.length ? result : null;
```

Comment:
if result.length == 0, then return blank array, and check length after you calling this function

Reply:
return result.length ? result : [];